lib/ansible/modules/remote_management/manageiq/manageiq_config.py

```diff
@@ -0,0 +1,211 @@
+        else:
+            dct[k] = merge_dct[k]
+
+
+def get_manageiq_config_value(module, name):
+    """ Gets the current value for the given config option.
+    :param module: AnsibleModule making the call
+    :param name: ManageIQ config option name
+    :return: dict of value of the ManageIQ config option
+    """
+    returncode, out, err = module.run_command([
+        "rails",
+        "r",
+        "puts MiqServer.my_server.get_config(:%s).config.to_json" % (name)
+    ], cwd=module.params['vmdb_path'])
+    if returncode != 0:
+        raise Exception("Error getting existing value for ':%s' config: %s" % (name, err))
+
+    return json.loads(out)
+
+
+def create_expected_value(original_value, changes):
+    """ Merges changes into original value to create a merged expected value.
+    :param original_value: dict original value to merge changes into
+    :param changes: dict of changes to merge into original_value to create the expected value
+    :return: dict of the changes merged into the original_value to create an expected value
+    """
+    expected_value = copy.deepcopy(original_value)
+    dict_merge(expected_value, changes)
+    return expected_value
+
+
+def main():
+    # define the module
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(type='str', required=True),
+            value=dict(type='dict', required=False, default={}),
+            vmdb_path=dict(type='str', required=False, default='/var/www/miq/vmdb'),
+        ),
+        supports_check_mode=True
+    )
+
+    # seed the result dict in the object
+    result = dict(
+        changed=False,
+        name='',
+        original_value={},
+        value={},
+        diff={}
+    )
+
+    # get the original value for the given config option
+    try:
+        original_value = get_manageiq_config_value(module, module.params['name'])
+    except Exception as err:
+        module.fail_json(msg=str(err), **result)
+
+    # update the result
+    result['name'] = module.params['name']
+    result['original_value'] = original_value
+
+    # create updated value dictionary
+    expected_value = create_expected_value(original_value, module.params['value'])
+
+    # enable diff mode
+    result['diff'] = {
+        'before': {
+            module.params['name']: original_value
+        },
+        'after': {
+            module.params['name']: expected_value
+        }
+    }
+
+    # if check_mode then stop here
+    if module.check_mode:
+        result['changed'] = original_value != expected_value
+        module.exit_json(**result)
+
+    # update config if difference
+    # else no-op
+    if original_value != expected_value:
+        (update_value_rc, update_value_out, update_value_err) = module.run_command([
+            "rails",
+            "r",
+            "MiqServer.my_server.set_config(:%s => JSON.parse('%s')); MiqServer.my_server.save!" % (module.params['name'], json.dumps(expected_value))
```

Comment:
I'm worried about Ruby quoting.  If any setting contains a single quote, this will break.
What I suggest is passing via ARGV.  `rails r` leaves in ARGV any args following the ruby fragment, e.g. this works:
```
# The \' escaping below is only to type this via shell;
# run_command() takes an array of strings and I think this way could pass arbitrary JSON with no other escaping at all.
rails r 'puts MiqServer.my_server.set_config(JSON.parse(ARGV.last))' '{"goodbye": "Qapla\', Worf"}'
```

Reply:
nifty trick. I like it.